### PR TITLE
feat(veeam-backup): order veeam step 2

### PR DESCRIPTION
**Pull request by chipp972**
ref: MANAGER-14896

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-14896 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/apps/veeam-backup/public/translations/order-veeam/Messages_fr_FR.json [hunk 1]
```diff
@@ -6,5 +6,16 @@
   "offer_content_part1": "Le service Veeam Data Platform dispose d'un plugin VCD pour sauvegarder les VM et les vApps depuis n'importe quel Virtual Data Center (VDC) de l'organisation.",
   "offer_content_part2": "Il est disponible au niveau de l'organisation pour tout utilisateur de VMware Cloud Director ayant le rôle d'administrateur de l'organisation.",
   "offer_price": "9e HT/VM/mois + consommation de OVHcloud Object storage",
-  "next_step": "Suivant"
+  "next_step": "Suivant",
+  "choose_org_title": "Choisissez une organisation VMware Cloud Director",
+  "no_organization_message": "Vous n'avez pas de service VMware Director actuellement. Si vous voulez en savoir plus sur notre produit Managed VMware Cloud Director, vous pouvez consulter notre page",
```
Comment (darsene) [inline packages/manager/apps/veeam-backup/public/translations/order-veeam/Messages_fr_FR.json:11]: Typo ?

```suggestion
  "no_organization_message": "Vous n'avez pas de service VMware Cloud Director actuellement. Si vous voulez en savoir plus sur notre produit Managed VMware Cloud Director, vous pouvez consulter notre page",
```

File: packages/manager/apps/veeam-backup/src/components/Links/VCDOrgInfoLink.component.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import { LinkType, Links, OvhSubsidiary } from '@ovhcloud/manager-components';
+
+const urls: Partial<{ [key in OvhSubsidiary]: string }> = {
+  FR: 'https://www.ovhcloud.com/fr/lp/vmware-vcd-evolution/',
+  GB: 'https://www.ovhcloud.com/en/lp/vmware-vcd-evolution/',
+  DEFAULT: 'https://www.ovhcloud.com/en/lp/vmware-vcd-evolution/',
```
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/components/Links/VCDOrgInfoLink.component.tsx:8]: As `GB` is the always default one, you can remove it

```suggestion
  DEFAULT: 'https://www.ovhcloud.com/en/lp/vmware-vcd-evolution/',
```

File: packages/manager/apps/veeam-backup/src/hook/date.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    const { result } = renderHook(() => useFormattedDate(null));
+    expect(result.current).toBe('1 janv. 1970');
+  });
+
+  it('displays a valid date if the value is valid', async () => {
+    const { result } = renderHook(() =>
+      useFormattedDate('2024-06-14T09:21:21.943Z'),
+    );
+    expect(result.current).toBe('14 juin 2024');
+  });
```
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/hook/date.spec.tsx:38]: **niptick** : can be group in only one test with` it.each`

File: packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts [hunk 4]
```diff
@@ -14,3 +24,6 @@ export const getOrganizationIdFromBackupId = (backupId = '') =>
 
 export const getOrganizationDisplayName = (org: VCDOrganization) =>
   org?.currentState?.fullName || org?.currentState?.name;
+
+export const getBackupIdFromOrganizationId = (orgId = '') =>
+  `${orgId}-veeam-backup`;
```
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts:29]: Is it really possible to not have any `orgId` here ?

File: packages/manager/apps/veeam-backup/src/pages/dashboard/SubscriptionTile.component.tsx [hunk 5]
```diff
@@ -46,13 +51,10 @@ export const SubscriptionTile: React.FC<VeeamBackupWithIam> = ({
...
-                <Description className="block">
-                  {formatDateString(
-                    data.data.billing.nextBillingDate,
-                    ovhLocaleToI18next(i18n.language),
-                  )}
-                </Description>
+              {nextBillingDate ? (
+                <Description className="block">{nextBillingDate}</Description>
+              ) : (
+                <OsdsSkeleton />
```
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/pages/dashboard/SubscriptionTile.component.tsx:57]: here, `isLoading` should be false so why display a skeleton ?

File: packages/manager/apps/veeam-backup/src/pages/order-veeam/OrderVeeamStep2.component.tsx [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+            onSortChange={setSorting}
+          />
+        )}
+        <div className="grid justify-items-center my-5">
+          {hasNextPage && (
+            <div>
+              <OsdsButton
+                color={ODS_THEME_COLOR_INTENT.info}
+                variant={ODS_BUTTON_VARIANT.stroked}
+                onClick={() => fetchNextPage()}
```
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/pages/order-veeam/OrderVeeamStep2.component.tsx:194]:
```suggestion
                onClick={fetchNextPage}
```

File: packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts [hunk 7]
```diff
@@ -1,8 +1,18 @@
 import { useQuery } from '@tanstack/react-query';
 import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import { useResourcesIcebergV2 } from '@ovhcloud/manager-components';
 import { getVcdOrganization } from '../api';
 import { VCDOrganization } from '../vcd.type';
 
+export const organizationListQueryKey = ['/vmwareCloudDirector/organization'];
+
+export const useOrgnizationList = ({ pageSize }: { pageSize?: number }) =>
```
Comment (pauldkn) [inline packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts:9]: **typo** `useOrganizationList`

File: packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts [hunk 8]
```diff
@@ -1,8 +1,18 @@
...
+  useResourcesIcebergV2<VCDOrganization>({
+    route: '/vmwareCloudDirector/organization',
+    queryKey: organizationListQueryKey,
+    pageSize,
+  });
+
 export const useOrganization = (organiationId: string) =>
```
Comment (pauldkn) [inline packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts:16]: **typo** `organizationId`

**Review by pauldkn — APPROVED** [review]

File: packages/manager/apps/veeam-backup/public/translations/order-veeam/Messages_fr_FR.json (hunk 1 above)
Comment (chipp972) [reply packages/manager/apps/veeam-backup/public/translations/order-veeam/Messages_fr_FR.json:11]: Bien vu

File: packages/manager/apps/veeam-backup/src/pages/dashboard/SubscriptionTile.component.tsx (hunk 5 above)
Comment (chipp972) [reply packages/manager/apps/veeam-backup/src/pages/dashboard/SubscriptionTile.component.tsx:57]: Yes right
I fixed it

File: packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts (hunk 4 above)
Comment (chipp972) [reply packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts:29]: Currently it is used only in one place and it's not possible there

File: packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/data/hook/useOrganizations.ts:1]: You've got lint errors

File: packages/manager/apps/veeam-backup/src/hook/date.spec.tsx
Comment (darsene) [inline packages/manager/apps/veeam-backup/src/hook/date.spec.tsx:1]: you've got lint error

**Pull request merged**